### PR TITLE
Shutdown Instances when license expires

### PR DESCRIPTION
**Pull request by hardillb**
part of #4074

## Description

- suspend all running projects when license expires
- prevent projects from being started
- audit log every day the license is expired

## Related Issue(s)

#4074

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by hardillb** [comment]
Need to work out how to write tests for this lot.

Can we mock `app.license`?

**Comment by knolleary** [comment]
A key part of #4074 is adding good notifications in the lead up to license expiry. That is the first item on the todo list, and following the discussion in that issue, something we need in place before we start shutting things down.

We cannot merge this part with out that piece in place as well.

**Comment by knolleary** [comment]
@hardillb this is still in draft - not sure the current state. Please see my comment above - we need to cover the notifications prior to expiry.

**Review by knolleary — DISMISSED** [review]
All looks good so far.

Just need the additional warnings prior to expiry before we can merge.

**Comment by hardillb** [comment]
@knolleary still draft because I wasn't sure we'd decided on just when to show the warnings. Will have another look when I've done with the nr-launcher file api and Steve's comments on the CSV logs

File: test/unit/forge/licensing/index_spec.js
Comment (hardillb) [inline test/unit/forge/licensing/index_spec.js:531]:
```suggestion
                await factory.createDevice(
```

**Review by joepavitt — APPROVED** [review]

**Pull request merged**
